### ISSUE TITLE
NancyHost initialization on Windows

### ISSUE THREAD

**Issue by metatheos** [OmniSharp/omnisharp-server#55]
After a few attempts I recently got YouCompleteMe to work on Windows 7. One issue I ran into is that it seems to launch multiple instances of Omnisharp that listen on different ports (which made the suggested workaround to use `urlacl` impractical).

While tracing down the source of the issue, I stumbled across <a href="http://blog.differentpla.net/post/34">this</a> post which suggests setting `HostConfiguration.RewriteLocalhost` to `false` in the `NancyHost` constructor.  Going by both the blog-post and the <a href="https://github.com/NancyFx/Nancy/wiki/Self-Hosting-Nancy#wiki-host-configuration">documentation</a> a `true` value for `RewriteLocalhost` (which happens to be the default one) makes `NancyHost` turn `localhost` into a <a href="http://msdn.microsoft.com/en-us/library/windows/desktop/aa364687(v=vs.85).aspx">strong wildcard domain</a>, which in turn requires elevated privileges.

Long story short, changing the constructor in `Program.cs` to

``` cs
var nancyHost = new NancyHost(new Bootstrapper(solution), new HostConfiguration{RewriteLocalhost=false}, new Uri("http://localhost:" + port));
```

(ycm is at ee198be6f06d39cef84b0817e0b88d3e08cbc477) worked for me.  Do I miss any reason for Omnisharp to not do this in general?
